util/chunk/chunk.go

```diff
@@ -210,6 +210,26 @@ func (c *Chunk) Reset() {
 	c.numVirtualRows = 0
 }
 
+// CopyTo copies this chunk to other.
+// If other is nil, create a new chunk.
+func (c *Chunk) CopyTo(other *Chunk) *Chunk {
+	if other == nil {
+		other = new(Chunk)
+	}
+	other.numVirtualRows = c.numVirtualRows
+	other.capacity = c.capacity
+	if len(other.columns) < len(c.columns) {
```

Comment:
It's only safe to call `CopyTo()` when the other `Chunk` shares the same schema as this one, `len(other.columns) < len(c.columns)` is an abnormal case.

In `prepareChunks()`, we actually need a copy constructor, how about just implementing a function `CopyConstruct() *Chunk`?

Reply:
`Chunk` doesn't store its schema information and it doesn't care about the schema, so we can't determine if these two `Chunks` have a same schema.
The meaning of `CopyTo` is ignoring their schema and copying data forcedly.

Reply:
> `Chunk` doesn't store its schema information

yes, it is.

> we can't determine if these two `Chunks` have a same schema.

no, it's not. We can do it in the caller side.

All the function implementations on Chunk haven't check whether it's correct to call these type related functions. The caller should promise that correctness. I think it's only meaningful when the two Chunk share the same schema to call this `CopyTo()` function.

As I mentioned in the above, I think what we actually need now is a copy constructor, for example `CopyConstruct() *Chunk`:
```go
func (c *Chunk) CopyConstruct() *Chunk {
	newChk := &Chunk{columns: make([]*column, len(c.columns)), numVirtualRows: c.numVirtualRows, capacity: c.capacity}
	for i := range c.columns {
		newChk.columns[i] = c.columns[i].CopyConstruct()
	}
	return newChk;
}
```

Reply:
ok, as your suggestion, `CopyConstruct` has been introduced and `CopyTo` has been removed. PTAL~